moveit_core/planning_scene/src/planning_scene.cpp

```diff
@@ -1491,8 +1471,16 @@ bool PlanningScene::processAttachedCollisionObjectMsg(const moveit_msgs::Attache
         // Transform shape poses to link frame
         if (object.object.header.frame_id != object.link_name)
         {
-          const Eigen::Isometry3d& transform = robot_state_->getGlobalLinkTransform(link_model).inverse() *
-                                               getTransforms().getTransform(object.object.header.frame_id);
+          const Eigen::Isometry3d& world_to_object_header_transform =
+              getTransforms().getTransform(object.object.header.frame_id);
+          Eigen::Isometry3d object_header_to_pose_transform;
+                                                          
+          PlanningScene::poseMsgToEigen(object.object.pose, object_header_to_pose_transform);
+          const Eigen::Isometry3d& world_to_object_origin_transform =
+              world_to_object_header_transform * object_header_to_pose_transform;
+
+          const Eigen::Isometry3d& transform =
```

Comment:
Same comment as below on this one, suggest dropping the reference here.

Reply:
Done